### PR TITLE
Changes to FV3 for ccpp-physics #782

### PR DESCRIPTION
**Pull request by dustinswales**
## Description
This PR contains changes to the RRTMGP code in the ccpp-physics. Coding errors were uncovered using the coupled ORTs.
Additionally there is some code (and .meta) cleanup 

There were two errors.
-) Indexing bug in SW cloud-sampling routine. (https://github.com/dustinswales/ccpp-physics/commit/0a545c622e92e20a9832b72f22238570dcc9c8bd#diff-ab383f7b24008085d38216e27611d9db32967aa782caf00525886f1c46407665L102)
-) Revert bug introduced by https://github.com/NCAR/ccpp-physics/pull/627. A little background. In this PR, the surface albedo calculation was moved out of the SW scheme and to the start of the radiation loop. In RRTMGP, the calculation of the solar-zenith angle was not moved out of the scheme along with the surface-albedo calculation. The SZA is an input to the subroutine to compute the surface-albedo.

## Testing
These changes were tested on Hera using the Intel compilers. The changes were motivated by ORT failures. Additional ORTs are proposed in the accompanying UFS PR.

## Dependencies
- waiting on https://github.com/NCAR/ccpp-physics/pull/782

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@dustinswales We'll be committing PR [921](https://github.com/ufs-community/ufs-weather-model/pull/921) this morning. Since Hera is down for maintenance tomorrow (12/7) and there is also a prod switch, we'd like to start the commit for UFS today if possible for your RRTMGP updates and new test. Would you be ready? I haven't seen a new PR yet at ufs-weather-model.

**Comment by dustinswales** [comment]
@DeniseWorthen 
Sorry for the delay, creating the PR's now.

**Comment by climbfuji** [comment]
@dustinswales I merged https://github.com/NCAR/ccpp-physics/pull/782, new ccpp-physics hash is cbc7e36
.

**Review by climbfuji — APPROVED** [review]
I can't comment on the new suite definition files, in particular the P8 file. All other changes look good. Need to revert `.gitmodules` and update the ccpp-physics submodule pointer.

I noticed that there is an update in the dycore, too, that is not documented in the PR. Is that intentional? Even if not, I don't think that it causes any harm (https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/compare/d6bfc1573d34cb2df20fe658568d9cca5232565d...fa86482e48f1d5b594acb369e68b8488de84dc66).

File: ccpp/suites/suite_FV3_GFS_v16_p8.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<suite name="FV3_GFS_v16_p8" version="1">
```
Comment (junwang-noaa) [inline ccpp/suites/suite_FV3_GFS_v16_p8.xml:3]: This may need to be updated later.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**